docs/arch/04-secrets-management.md

```diff
@@ -0,0 +1,183 @@
+# Secrets Management
+
+ToolHive provides a secrets management system for securely handling API keys, tokens, and other sensitive data needed by MCP servers.
+
+## Architecture
+
+```mermaid
+graph LR
+    subgraph "Providers"
+        Encrypted[Encrypted Storage<br/>AES-256-GCM]
+        OnePass[1Password CLI]
+        Env[Environment Vars]
+    end
+
+    Provider[Secret Provider] --> Fallback[Fallback Chain]
+    Encrypted --> Provider
+    OnePass --> Provider
+    Env --> Provider
+    Fallback --> Container[Container EnvVars]
+
+    Keyring[OS Keyring] -.->|password| Encrypted
+
+    style Encrypted fill:#81c784
+    style Keyring fill:#ba68c8
+```
+
+## Provider Types
+
+**Implementation**: `pkg/secrets/types.go`
+
+### 1. Encrypted (Default)
+
+- **Storage**: `~/.local/share/toolhive/secrets_encrypted` (Linux)
+- **Encryption**: AES-256-GCM
+- **Password**: Stored in OS keyring (keyctl/Keychain/DPAPI)
+- **Capabilities**: Read, write, delete, list
+
+**Implementation**: `pkg/secrets/encrypted.go`
+
+### 2. 1Password
+
+- **Storage**: 1Password vaults
+- **Access**: Via 1Password SDK (`github.com/1password/onepassword-sdk-go`)
+- **Authentication**: Service account token (`OP_SERVICE_ACCOUNT_TOKEN`)
+- **Capabilities**: Read-only, list
+
+**Implementation**: `pkg/secrets/1password.go`
+
+### 3. Environment
+
+- **Storage**: Environment variables (`TOOLHIVE_SECRET_*`)
+- **Use case**: CI/CD, stateless deployments
+- **Capabilities**: Read-only
```

Comment:
The Environment provider capabilities description is incomplete. While it is read-only, it's important to note that `ListSecrets` is **explicitly disabled for security reasons** to prevent enumeration of environment variables.

```suggestion
- **Capabilities**: Read-only (ListSecrets explicitly disabled for security)
```

Evidence: `pkg/secrets/environment.go` lines 54-56 show that ListSecrets returns an error intentionally